xds/internal/httpfilter/httpfilter.go

```diff
@@ -98,5 +99,12 @@ func Register(b Filter) {
 //
 // If no filter is register with typeURL, nil will be returned.
 func Get(typeURL string) Filter {
+	// Hardcoded here vs. rbac http filter package because rbac HTTP Filter test
+	// is in integration test.
+	if !env.RBACSupport {
+		if typeURL == "type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBAC" || typeURL == "type.googleapis.com/envoy.extensions.filters.http.rbac.v3.RBACPerRoute" {
+			return nil
+		}
+	}
```

Comment:
Yeah, I agree that it's much cleaner to conditionally register the filter, such as like what you had in fault. However, the problem comes from the fact that my test for this filter is in server_integration_test.go, which has a lot of helper functions. In order to test this filter (and actually build the filter) in the integration test, you need to have some way of getting past this check. Since init() runs in the imported rbac package before the test runs, there is no way to register the builder{} struct, similar to how you did it in fault injection. I tried moving the test over, but it created a ton of refactoring work and moving hundreds of lines of methods around, and I found this to be a more acceptable/cleaner compromise, no matter how ugly/hacky the one line seems to be. What do you think?

Reply:
Maybe I'm not following, but what about something like `httpfilter.UnregisterForTesting(typeURL string)`?

Reply:
Yeah, but then you have no way of toggling it on in a test because httpfilter init() runs protected by an environment variable, are you suggesting adding a way to register the filter mid test/deregister the filter mid test?

Reply:
> are you suggesting adding a way to register the filter mid test/deregister the filter mid test?

True, we need to re-register, too.  So maybe:

```go
httpfilter.UnregisterForTesting(typeURL string)
rbac.UnregisterForTesting() // calls above
rbac.RegisterForTesting() // calls httpfilter.Register regardless of env var.
```

?

Reply:
I disagree with this. This is adding a lot of stuff to an API to simply get rid of this Global check. Unregistering/Deregistering the filter is somewhat orthogonal to testing an environment variable protecting functionality. We are testing that env variable atomically, not a small piece of functionality such as registering the filter.

Reply:
Adding special cases to otherwise completely generic code should be avoided wherever possible.

The more natural way to do this would be to conditionally register the RBAC filter at startup and be done.  But that makes testing with and without the filter registered impossible, so we can't do that.  So it seems the next best thing is to add some testing-only functions that allow us to set up the state as we need for the tests.  If you have other suggestions for doing that, I'm fine with other options, but let's not put special cases into very fundamental, filter-agnostic code for testing purposes unless it's a last resort.

Reply:
Yeah, I see what you're saying. Makes total sense. Added the three functions, Unregister/Register in RBAC, and also Unregister in HTTP Filter. There's a test breakage that this next commit will have, but it's also checked into master (in csds.go).